src/status_im/communities/core.cljs

```diff
@@ -457,50 +456,17 @@
   [{:keys [db]}]
   {:db (assoc db :communities/create-channel {})})
 
-(rf/defn invite-people-pressed
-  {:events [:communities/invite-people-pressed]}
-  [cofx id]
-  (rf/merge cofx
-            (reset-community-id-input id)
-            (bottom-sheet/hide-bottom-sheet-old)
-            (navigation/open-modal :invite-people-community {:invite? true})))
-
-(rf/defn share-community-pressed
-  {:events [:communities/share-community-pressed]}
-  [cofx id]
-  (rf/merge cofx
-            (reset-community-id-input id)
-            (bottom-sheet/hide-bottom-sheet-old)
-            (navigation/open-modal :invite-people-community {})))
-
-(rf/defn create-channel-pressed
-  {:events [::create-channel-pressed]}
-  [{:keys [db] :as cofx} id]
-  (rf/merge cofx
-            (reset-community-id-input id)
-            (reset-channel-info)
-            (rf/dispatch [::create-channel-fields
-                          (rand-nth emoji-thumbnail-styles/emoji-picker-default-thumbnails)])
-            (navigation/open-modal :create-community-channel {:community-id id})))
-
-(rf/defn edit-channel-pressed
-  {:events [::edit-channel-pressed]}
-  [{:keys [db] :as cofx} community-id chat-name description color emoji chat-id category-id position]
-  (let [{:keys [color emoji]} (if (string/blank? emoji)
-                                (rand-nth emoji-thumbnail-styles/emoji-picker-default-thumbnails)
-                                {:color color :emoji emoji})]
-    (rf/merge cofx
-              {:db (assoc db
-                          :communities/create-channel
-                          {:name            chat-name
-                           :description     description
-                           :color           color
-                           :community-id    community-id
-                           :emoji           emoji
-                           :edit-channel-id chat-id
-                           :category-id     category-id
-                           :position        position})}
-              (navigation/open-modal :edit-community-channel nil))))
+(re-frame/reg-event-fx :communities/invite-people-pressed
+ (fn [{:keys [db]} [id]]
+   {:db (assoc db :communities/community-id-input id)
+    :fx [[:dispatch [:hide-bottom-sheet]]
+         [:dispatch [:open-modal :invite-people-community {:invite? true}]]]}))
```

Comment:
im not sure about this, why :fx is separated, because db is also fx, so it should be just replacement by vector, so

```clojure
(fn [...] 
   [ [:db (...)]
     [:dispatch]
     [:dispatch]]
```

Reply:
if we want to use this then it should be done in another way, like
```clojure
(defn invite-people-pressed []
    (dispatch [:communities/invite-people-pressed])
    (dispatch [:hide-bottom-sheet])
    (dispatch [:open-modal])
```

Reply:
not sure I follow here, 
while maybe you are right that `:db` is an `fx`, re-frame docs indicates that the "canonical" way would be using 
```
{:db ...
   :fx [..]}
```
So the above looks correct to me (and I guess anyone who comes from the docs).

The function 
```
(defn invite-people-pressed []
    (dispatch [:communities/invite-people-pressed])
    (dispatch [:hide-bottom-sheet])
    (dispatch [:open-modal])
```
looks different, as that's actually firing the dispatch, so it would have side effects, but maybe I am misunderstanding what you mean?

Reply:
> while maybe you are right that `:db` is an `fx`, re-frame docs indicates that the "canonical" way would be using

yes, i just say that re-frame not always give good "canonical" way, so for me its just confusing, why we have :fx better to just change map to the vector, so it's a question more for re-frame, not to how its used in our code




> but maybe I am misunderstanding what you mean?

so what i want to say `:dispatch` effect should be used, so when i dispatch `:communities/invite-people-pressed` im sure it doesn't dispatch anything, but i agree moving it to the action handler function also not a good idea at some point of view , so idk , just thinking loud :)

Reply:
I see, 
I think there's value in keeping it as vanilla as possible, especially for new comers/people who might not be familiar with the app already, as they most likely will refer back to the docs, and the discrepancy might add some attrition, even though some of the decision might be questionable

Reply:
I agree with @flexsurfer about re-frame's API. :thinking: 
The API hasn't evolved in  the best way, at the beggining I was wondering "why do I have to always use `:dispatch` inside each vector of `:fx`? 

`:fx` was a later fix in re-frame and it looks like a little forced in terms of API.

But same as @cammellos , I prefer having it as vanilla as possible, adding custom macros that do complex things  or define complex patterns add an extra layer of complexity.